### DISCUSSION TITLE
Why does ClusterPipeline not provide the syncAndReturnAll() method in version 5.0.0 beta 2?

### DISCUSSION THREAD

**Discussion by sundays0** [xetorthio/jedis#3516] [Q&A] [answered]
I am currently trying to use version 5.0.0 beta 2 and have found that the ClusterPipeline class returned through JedisCluster's pipelined() method does not have the syncAndReturnAll() method, which makes some operations incompatible with classes returned through JedisPool's pipelined() method. What is the reason for this?

I am now preparing to abstract and package JedisPooled, JedisSentineled, and JedisCluster, which can shield Redis deployment differences from business layer code. Due to the above issue, I was unable to provide a sync() method with a return value when abstracting a Pipeline;

By the way, in JedisSentineled, seems to have forgotten to rewrite the pipelined() method? Currently, only the PipelineBase object can be obtained.

**Comment by sazzad16** [comment, marked as answer]
> Why does ClusterPipeline not provide the syncAndReturnAll() method

Because the returned list would have no logical order (with current implementation).
The implementation of syncAndReturnAll() with a logical order at our end will add a processing cost for all user applications which is not desired by many users.

Solution: You can maintain a List of `Response`s in your code. After `sync()`, generate the List of actual replies from that List.

**Reply by sundays0** [reply]
Thank you! 

Also, should the pipelined () method be overridden in JedisSentinered?

**Reply by sazzad16** [reply]
> should the pipelined () method be overridden in JedisSentinered?

done
